packages/snaps-execution-environments/src/common/endowments/network.ts

```diff
@@ -121,9 +171,26 @@ const createNetwork = () => {
       );
     }
 
+    let started = false;
+    const onStart = () => {
+      if (!started) {
+        started = true;
+        notify({ method: 'OutboundRequest' });
+      }
+    };
+
+    let finished = false;
+    const onFinish = () => {
+      if (!finished) {
+        finished = true;
+        notify({ method: 'OutboundResponse' });
+      }
+    };
+
     let res: Response;
     let openFetchConnection: { cancel: () => Promise<void> } | undefined;
     try {
+      notify({ method: 'OutboundRequest' });
```

Comment:
I'm thinking if it's reasonable to notify and pause the timer for every `fetch` call 🤔 
Or maybe to optimize it to be used only when fetching is taking too long.

I think that this might put more pressure to the _RPC / Post Message Stream_ communication channels if it's done for every `fetch` call. I imagine use case where a Snap has many `fetch` calls during it's runtime (request handling). For example, I can use fetch to get 100 svg icons of a tokens which I'm going to show inside Snap Home / Custom UI, etc. That would be a lot of redundant notifications going on and timer pause actions handled, given that fetch for use case like that would complete in a second or two approximatelly.

I would think of making heuristic function that would do notify/pause actions only if it gets critical for the fetch to complete in time. What would be critical is a good question, but we know that each request has 60 seconds to complete before the timeout, let's say that the long-running one is the one that takes more than 30 seconds or similar. Something that is not too close to the max runtime (60 seconds) and would allow timer pausing to complete in time.

Reply:
We already do this for every request to `snap.request` and `ethereum.request` without problems. I don't think we need to be worried about the RPC communication channels. 

We could only trigger the notification when an async function is called on the response, e.g. `json()` if we wanted to only notify for "heavy" operations. But the fetch call itself may run a little long if the initial HTTP request takes a long time to initiate.

Reply:
Resolved in other conversations on Slack. Identified as minor concern.